process_jartic_archives_fast.py

```
@@ -0,0 +1,217 @@
+        
+        # Split locations into batches
+        batch_size = max(1, len(locations) // self.max_workers)
+        location_batches = [
+            locations[i:i + batch_size] 
+            for i in range(0, len(locations), batch_size)
+        ]
+        
+        # Progress bar for batches
+        batch_progress = tqdm(
+            total=len(location_batches),
+            desc=f"📦 Processing batches",
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)